OpenGov-OpenData/ckanext-og_datatablesview
# [OD-1645] Add Support for CKAN 2.9

## [OD-1645](https://opengovinc.atlassian.net/browse/OD-1645)

## Description

This PR adds support for CKAN 2.9 while also being backward compatible for older CKAN version.

**Important Changes:**
- `ckanext/og_datatablesview/plugin/` directory was added to split when to use pylons or flask for appropriate CKAN version.
- ckanext/og_datatablesview/public` was renamed to `ckanext/og_datatablesview/assets/` to support adding assets for flask. Along with this `ckanext/og_datatablesview/assets/webassets.yml` file was added to generate these assets
- Logic for controller/route was now moved to `ckanext/og_datatablesview/utils.py`
- Unit Tests were refactored to use pytest.

## Manual Test Procedure

NOTE: This manual test should be done for CKAN 2.9 and CKAN 2.7
1. Clone this directory to where your plugins are stored on your local and checkout this branch ``
2. In this directory, run `pip install -r requirements.txt` and then run `python setup.py develop`
3. Haven `og_datatablesview` enabled in your ini file, `og_datatablesview`
4. Start CKAN and create a new dataset with a resource that contains a blank field. NOTE: Ensure that the column where the blank field will be contains only numbers.
5. Create "OG Data Tables" view for the newly created resource.
6. Ensure you see options "Hide Export" and. "Display Copy and Print buttons".
7. Create view and then check a blank cell is displayed where the empty cell is.
8. Play around with filtering data to ensure that functionality still works.
9. Click the "Export" button make sure you are able to download the data.
10. Edit view and toggle options mentioned int Step 6. to ensure they are working.

## Unit Test
In directory run `pytest --disable-warnings`

## Approval Criteria 

- Remove `None` values when the data type of the column is set to numeric. Data types of columns can be configured using the data dictionary tab when editing resources. I've attached a video showing an example of updating data types with the data dictionary 
- Rename Downloads button to Export and make the display of the button configurable in view edit page. The config can be called “Hide Export button“
- Rename current config for “Display export buttons” to “Display Copy and Print buttons“

## Submitter Checklist
